runtime/util/annhelp.c

```diff
@@ -139,3 +143,205 @@ skipAnnotationElement(J9ROMConstantPoolItem const * constantPool, U_8 const *dat
+{
+	BOOLEAN annotationFound = FALSE;
+	J9ROMClass *romClass = NULL;
+	J9ROMConstantPoolItem *constantPool = NULL;
+	J9ROMMethodRef *romMethodRef = NULL;
+	J9ROMMethod *romMethod = NULL;
+
+	Assert_VMUtil_true(NULL != clazz);
+	Assert_VMUtil_true(NULL != annotationName);
+
+	romClass = clazz->romClass;
+	constantPool = J9_ROM_CP_FROM_ROM_CLASS(clazz->romClass);
+	romMethodRef = (J9ROMMethodRef *) &constantPool[cpIndex];
+	romMethod = getMethodFromMethodRef(romClass, romMethodRef);
+
+	if (NULL != romMethod) {
+		U_32 *methodAnnotationData = getMethodAnnotationsDataFromROMMethod(romMethod);
+
+		if (NULL != methodAnnotationData) {
+			U_32 len = *methodAnnotationData;
+			U_8 *data = (U_8 *) (methodAnnotationData + 1);
+
+			annotationFound = findRuntimeVisibleAnnotation((U_8*)data, len, annotationName, constantPool);
+		}
+	}
+
+	Trc_Util_annhelp_SearchForMethodAnnotation(J9UTF8_DATA(annotationName), cpIndex, clazz, romMethod, (UDATA) annotationFound);
+
+	return annotationFound;
+}
+
+/**
+ * Find the corresponding J9ROMMethod for a given J9ROMMethodRef.
+ *
+ * @param romClass The ROM Class the method belongs to.
+ * @param romMethodRef The Method Ref to find the matching Method Shape for.
+ * @return A pointer to the corresponding Method Shape or NULL if not found.
+ */
+static J9ROMMethod *
+getMethodFromMethodRef(J9ROMClass *romClass, J9ROMMethodRef *romMethodRef)
+{
+	J9ROMMethod *romMethod = NULL;
+	J9ROMNameAndSignature *methodRefNameAndSig = J9ROMMETHODREF_NAMEANDSIGNATURE(romMethodRef);
+	J9UTF8 *methodRefName = J9ROMNAMEANDSIGNATURE_NAME(methodRefNameAndSig);
+	J9UTF8 *methodRefSignature = J9ROMNAMEANDSIGNATURE_SIGNATURE(methodRefNameAndSig);
+	J9ROMMethod *romMethodPtr = J9ROMCLASS_ROMMETHODS(romClass);
+	U_32 i;
+
+	for (i = 0; (i < romClass->romMethodCount) && (NULL != romMethodPtr); i++) {
+		J9UTF8 *methodName = J9ROMMETHOD_NAME(romMethodPtr);
+		J9UTF8 *methodSignature = J9ROMMETHOD_SIGNATURE(romMethodPtr);
+
+		if (J9UTF8_EQUALS(methodRefName, methodName) && J9UTF8_EQUALS(methodRefSignature, methodSignature)) {
+			romMethod = romMethodPtr;
+			break;
+		}
+
+		romMethodPtr = nextROMMethod(romMethodPtr);
+	}
+
+	return romMethod;
+}
+
+/**
+ * Check if the provided Runtime Visible annotation data contains the specified annotation.
+ *
+ * @param data The Runtime Visible annotation data.
+ * @param annotationName The annotation to check for.
+ * @param constantPool The constant pool where the data is located.
+ * @return TRUE if the annotation is found, FALSE otherwise.
+ */
+static BOOLEAN
+findRuntimeVisibleAnnotation(U_8 *data, U_32 length, J9UTF8 *annotationName, J9ROMConstantPoolItem *constantPool)
+{
+	U_8 *dataEnd = (U_8*)data + length;
+	U_32 errorCode = 0; /* used by CHECK_EOF */
+	U_32 offset = 0; /* used by CHECK_EOF */
+	U_8 const *index = data; /* used by CHECK_EOF */
+	U_32 attributeLength = 0;
+	U_32 numAnnotations = 0;
+	I_32 getAnnotationResult = -1;
+	BOOLEAN res = FALSE;
```

Comment:
use complete word `result` or `found`
```suggestion
	BOOLEAN found = FALSE;
```